crates/modelardb_server/tests/integration_test.rs

```diff
@@ -736,3 +761,27 @@ fn test_optimized_query_results_equals_non_optimized_query_results() {
 
     assert_eq!(optimized_query, non_optimized_query);
 }
+
+#[test]
+fn test_cannot_update_invalid_setting() {
+    let mut test_context = TestContext::new();
+    let response = test_context.update_configuration("invalid", "1");
+
+    assert!(response.is_err());
+    assert_eq!(
+        response.err().unwrap().message(),
+        "invalid is not a valid setting in the server configuration."
+    );
+}
+
+#[test]
+fn test_cannot_update_setting_with_invalid_value() {
```

Comment:
If additional checks are add based on the comments above, maybe add more tests to check they work as intended.

Reply:
I have not added any additional checks yet since I think the parse to `usize` handles what should be checked. Do you think we should restrict the upper bound of the reserved memory so the restriction is lower than the max of `usize`?

Reply:
I think it is fine to keep the value as any `usize` value as no matter what limit we set it will cause problems. If we choose a static "reasonable" max size it will have to be large enough for high-end server, thus it would not prevent users from setting the memory limited higher than the machine can reliably handle. If we use a value based on the amount of [free memory](https://docs.rs/sysinfo/latest/sysinfo/trait.SystemExt.html#tymethod.free_memory) we will unnecessarily limit the amount of memory `modelardbd` can be asked to allocate as some of the allocated memory may be available for reuse, while limiting it to the [total amount of memory](https://docs.rs/sysinfo/latest/sysinfo/trait.SystemExt.html#tymethod.total_memory) in the machine would not prevent the user from allocating too much memory to `modelardbd` as some memory is needed for the OS and other applications. Limiting it to the current amount of [available memory](https://docs.rs/sysinfo/latest/sysinfo/trait.SystemExt.html#tymethod.available_memory) could be a solution, but that is apparently not possible on Microsoft Windows and FreeBSD. What do you think @chrthomsen?

Reply:
I would be temped to leave it to the user to pick a reasonable value, in particular if the behavior of those functions is inconsistent across platforms.